tests/datasets/test_eudem.py

```diff
@@ -33,22 +34,22 @@ def test_getitem(self, dataset: EUDEM) -> None:
         assert isinstance(x["mask"], torch.Tensor)
 
     def test_extracted_already(self, dataset: EUDEM) -> None:
-        assert isinstance(dataset.paths, str)
-        zipfile = os.path.join(dataset.paths, "eu_dem_v11_E30N10.zip")
-        shutil.unpack_archive(zipfile, dataset.paths, "zip")
+        assert check_instance_type(dataset.paths)
```

Comment:
I think we can just assert that this is a `pathlib.Path`

Reply:
@adamjstewart  We may not be able to do that because we have to assert for ``str``,``bytes`` , ``os.PathLike[str]`` and
``os.PathLike[bytes]``.